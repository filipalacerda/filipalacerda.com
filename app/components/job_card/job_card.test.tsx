import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import "@testing-library/jest-dom";

import JobCard from "./index";
import jobs from "../../resume/data";

describe(JobCard.name, () => {
  let container: HTMLElement;

  const job = jobs[0];

  beforeEach(() => {
    container = render(
      <JobCard
        company={job.company}
        title={job.title}
        date={job.date}
        description={job.description}
        location={job.location}
      />,
    ).container;
  });

  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("should render company name", () => {
    expect(screen.getByText(job.company.name)).toBeInTheDocument();
  });

  it("should render title", () => {
    expect(screen.getByText(job.title)).toBeInTheDocument();
  });

  it("should render date", () => {
    expect(screen.getByText(job.date.from)).toBeInTheDocument();
    expect(screen.getByText(job.date.to)).toBeInTheDocument();
  });

  it("should render the logo", () => {
    expect(
      screen.getAllByAltText(`${job.company.name} logo`),
    ).toBeInTheDocument();
  });

  it("should not render description by default", () => {
    expect(screen.getByText(job.description)).not.toBeInTheDocument();
  });

  it("should render description when the user clicks read more", async () => {
    await userEvent.click(screen.getByRole("button"));
    expect(screen.getByText(job.description)).toBeInTheDocument();
  });
});
