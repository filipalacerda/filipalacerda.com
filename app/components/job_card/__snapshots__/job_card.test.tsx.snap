// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard should match snapshot 1`] = `
<div>
  <div
    class="border border-darkBlue rounded flex py-2 px-2 flex flex-col mt-4"
  >
    <div
      class="section-top flex flex-col"
    >
      <div
        class="card flex justify-between"
      >
        <div
          class="header"
        >
          <h1
            class="text-2xl font-light underline text-green"
          >
            <a
              href="https://livelyme.com/"
            >
              Lively
            </a>
          </h1>
          <h2
            class="text-lg font-bold mt-2"
          >
            Senior Frontend Engineer
          </h2>
          <p
            class="font-light text-xs mt-2"
          >
            May 2024
             - 
            Present
          </p>
          <span
            class="font-light text-xs mt-2"
          >
            Remote
          </span>
        </div>
        <div
          class="mt-2.5 mr-3.5"
        >
          <img
            alt="Lively logo"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
            style="color: transparent;"
            width="100"
          />
        </div>
      </div>
      <button
        aria-label="Read more"
      >
        <span
          aria-hidden="true"
          aria-label="Toggle description icon"
          class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
